pkg/v1/sdk/features/runner/runner.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package runner
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+)
+
+// FeatureGateRunner defines methods to interact with FeatureGate resources
+type FeatureGateRunner struct {
+	FeatureGateClient client.Client
+}
+
+// NewFeatureGateRunner returns an instance of featureGateRunner.
+func NewFeatureGateRunner() (*FeatureGateRunner, error) {
```

Comment:
optional: If someone is using this programmatically, they might be interested in passing their own `client.Client` instead of a client obtained from `getFeatureGateClient`. Please consider either changing this function or splitting into two constructors (one with default client and one which uses a passed in client).

Okay to revisit later as well.

Reply:
i did think about this, but thought will add it later if needed, but again I thought let's add it now :) I took functional options approach so that we can grow this API over time if needed and its more readable https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis